torch/csrc/jit/passes/to_batch.cpp

```diff
@@ -0,0 +1,78 @@
+#include "torch/csrc/jit/passes/to_batch.h"
+#include "torch/csrc/jit/script/compiler.h"
+
+namespace torch { namespace jit {
+
+// map from batchTensor to {data, mask, dims}
+static std::unordered_map<Value*, std::vector<Value*>> batch_map;
```

Comment:
This shouldn't be a static global variable. It will introduce race conditions and leak memory.  Instead, it is best to put context like this into a class, and have functions like `ToBatch` be a method of that class.